vanityssh.go

```diff
@@ -120,19 +126,29 @@ func main() {
 
 	deleteLine := "\033[2K\r"
 	cursorUp := "\033[A"
-	prev_counter := global_counter
-	prev_time := time.Now()
+	avgKeyRate := float64(global_counter)
+	oldCounter := global_counter
+	oldTime := time.Now()
 
 	for {
 		time.Sleep(250 * time.Millisecond)
+		relTime := time.Now().Sub(oldTime).Seconds()
 
-		relTime := time.Now().Sub(prev_time).Seconds()
+		// on first run, initialize the moving average with the current rate
+		// instead of starting at 0 and taking many seconds to tend towards the
+		// actual key rate
+		if oldCounter == 0 {
+			avgKeyRate = float64(global_counter)
```

Comment:
every few runs, this hits a race condition and stays at 0. i figure this is fine, considering after a few seconds it tends towards the actual number anyways